### PR TITLE
Better Loss Plot

### PR DESCRIPTION
**Pull request by jerrymhuang**
bf_new_loss

Fixed and refined `bf.diagnostics.plots.loss`:

* Replace the moving average with an exponential moving average and make it functional and by default;
* Provide the user with a new set of customization, including moving average window (span), color maps, markers, and grid;
* Make the validation loss a viridis gradient line.

This version of loss plot also propagates to the `workflow` object.

### CODE REVIEW

File: bayesflow/diagnostics/plots/loss.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from collections.abc import Sequence
+from typing import Sequence
```
Comment (LarsKue) [inline bayesflow/diagnostics/plots/loss.py:1]: We should prefer `collections.abc`.

File: bayesflow/diagnostics/plots/loss.py [hunk 2]
```diff
@@ -7,21 +7,25 @@
...
 
 def loss(
     history: keras.callbacks.History,
     train_key: str = "loss",
     val_key: str = "val_loss",
-    moving_average: bool = False,
-    per_training_step: bool = False,
-    ma_window_fraction: float = 0.01,
+    moving_average: bool = True,
+    moving_average_alpha: float = 0.8,
```
Comment (LarsKue) [inline bayesflow/diagnostics/plots/loss.py:18]: Calling this `alpha` is a bit confusing with respect to opacity. I would prefer `smoothing_factor` or `moving_average_smoothing_factor`.

**Review by LarsKue — CHANGES_REQUESTED** [review]
Looks great! I will add the changes that I proposed myself.

**Comment by LarsKue** [comment]
Although it is very pretty, I am removing the gradient line for now, until we have resolved how we can reliably add it to the legend. I also simplified the rest of the code and fixed an issue with the smoothing factor being applied the wrong way around.

**Review by LarsKue — APPROVED** [review]
